### PR TITLE
Add real_time to solver_specific output

### PR DESCRIPTION
**Pull request by abelsiqueira**
Based on our meeting today.

### CODE REVIEW

File: src/NLPModelsIpopt.jl [hunk 1]
```diff
@@ -150,7 +152,10 @@ function ipopt(nlp :: AbstractNLPModel;
                                objective=problem.obj_val, dual_feas=dual_feas, iter=iter,
                                primal_feas=primal_feas, elapsed_time=Δt, multipliers=problem.mult_g,
                                multipliers_L=problem.mult_x_L, multipliers_U=problem.mult_x_U,
-                               solver_specific=Dict(:internal_msg => Ipopt.ApplicationReturnStatus[status])
+                               solver_specific=Dict(
+                                 :internal_msg => Ipopt.ApplicationReturnStatus[status],
+                                 :real_time => real_time
```
Comment (dpo) [inline src/NLPModelsIpopt.jl:157]: @tmigot Could we do something like this for Knitro too?

Comment (tmigot) [reply src/NLPModelsIpopt.jl:157]: Good idea, I will update the ongoing PR. Thanks

**Comment by tmigot** [comment]
I have also noticed for NLPModelsKnitro that  `cpu_time < real_time` is not always verified as shown by the unit tests. Not sure why though.
What about making the following test?

```
(stats, t) = @timed ipopt(nlp)
 @test t[1] ≥ stats.solver_specific[:real_time]
```

**Comment by abelsiqueira** [comment]
Do you mean changing `>` to `>=` here: https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/pull/52/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R8?
Also, what is the difference between using `@timed` instead of `time()`?

**Comment by tmigot** [comment]
> Do you mean changing `>` to `>=` here: https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/pull/52/files#diff-3b9314a6f9f2d7eec1d0ef69fa76cfabafdbe6d0df923768f9ec32f27a249c63R8?

Actually, no. I have noticed on Knitro's logs that the CPU time can be larger than the real time, which also seems to be the error in the unit test. I don't know why this is the case though, but it seems independent of the package.

> Also, what is the difference between using `@timed` instead of `time()`?
I believe it is the same, with @timed, we just avoid defining a start_time before the execution and check the difference afterwards.

**Comment by abelsiqueira** [comment]
I removed the test. Should be good to go now.

**Comment by dpo** [comment]
Thank you.

**Pull request merged**